### PR TITLE
[SPARK-37262][SQL] Don't log empty aggregate and group by in JDBCScan

### PR DESCRIPTION
**Pull request by huaxingao**
### What changes were proposed in this pull request?
Currently, the empty pushed aggregate and pushed group by are logged in Explain for JDBCScan
```
Scan org.apache.spark.sql.execution.datasources.v2.jdbc.JDBCScan$$anon$1@72e75786 [NAME#1,SALARY#2] PushedAggregates: [], PushedFilters: [IsNotNull(SALARY), GreaterThan(SALARY,100.00)], PushedGroupby: [], ReadSchema: struct<NAME:string,SALARY:decimal(20,2)>
```

After the fix, the JDBCSScan will be
```
Scan org.apache.spark.sql.execution.datasources.v2.jdbc.JDBCScan$$anon$1@72e75786 [NAME#1,SALARY#2] PushedFilters: [IsNotNull(SALARY), GreaterThan(SALARY,100.00)], ReadSchema: struct<NAME:string,SALARY:decimal(20,2)>
```

### Why are the changes needed?
address this comment https://github.com/apache/spark/pull/34451#discussion_r740220800

### Does this PR introduce _any_ user-facing change?
no

### How was this patch tested?
existing tests

### CODE REVIEW

**Comment by huaxingao** [comment]
cc @c21 @cloud-fan

File: sql/core/src/main/scala/org/apache/spark/sql/execution/DataSourceScanExec.scala [hunk 1]
```diff
@@ -151,9 +144,10 @@ case class RowDataSourceScanExec(
 
     Map(
       "ReadSchema" -> requiredSchema.catalogString,
-      "PushedFilters" -> seqToString(markedFilters.toSeq),
-      "PushedAggregates" -> aggString,
-      "PushedGroupby" -> groupByString) ++
+      "PushedFilters" -> seqToString(markedFilters.toSeq)) ++
+      pushedDownOperators.aggregation.fold(Map[String, String]()) { v =>
+        Map("PushedAggregates" -> seqToString(v.aggregateExpressions),
+          "PushedGroupby" -> seqToString(v.groupByColumns))} ++
```
Comment (c21) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/DataSourceScanExec.scala:150]: @sunchao had a nit comment to rename to `PushedGroupBy` in https://github.com/apache/spark/pull/34498#discussion_r744076628 . Though I feel `PushedGroupByColumns` are better. To me, it seems to mean the same thing between `aggregate` and `group by`.

File: sql/core/src/test/scala/org/apache/spark/sql/jdbc/JDBCV2Suite.scala [hunk 2]
```diff
@@ -344,8 +344,7 @@ class JDBCV2Suite extends QueryTest with SharedSparkSession with ExplainSuiteHel
       case _: DataSourceV2ScanRelation =>
         val expected_plan_fragment =
           "PushedAggregates: [MAX(ID), MIN(ID)], " +
-            "PushedFilters: [IsNotNull(ID), GreaterThan(ID,0)], " +
-            "PushedGroupby: []"
```
Comment (c21) [inline sql/core/src/test/scala/org/apache/spark/sql/jdbc/JDBCV2Suite.scala:348]: don't we print `PushedAggregates` and `PushedGroupby` together in `DataSourceScanExec.scala`? Why remove `PushedGroupby` only here?

**Comment by c21** [comment]
Looks okay to me besides some minor comments, thanks @huaxingao.

File: sql/core/src/test/scala/org/apache/spark/sql/jdbc/JDBCV2Suite.scala (hunk 2 above)
Comment (huaxingao) [reply sql/core/src/test/scala/org/apache/spark/sql/jdbc/JDBCV2Suite.scala:348]: The query is `select MAX(ID), MIN(ID) FROM h2.test.people where id > 0`. The aggregates `MAX(ID)` and `MIN(ID)` are pushed down. There is no group by, so pushed group by is [].

Comment (c21) [reply sql/core/src/test/scala/org/apache/spark/sql/jdbc/JDBCV2Suite.scala:348]: Yes I understand the PushedGroupby is empty here. I don't know why we need to remove it here. We can still test it to be empty, right?

File: sql/core/src/main/scala/org/apache/spark/sql/execution/DataSourceScanExec.scala (hunk 1 above)
Comment (huaxingao) [reply sql/core/src/main/scala/org/apache/spark/sql/execution/DataSourceScanExec.scala]: Changed to `PushedGroupByColumns`

File: sql/core/src/test/scala/org/apache/spark/sql/jdbc/JDBCV2Suite.scala (hunk 2 above)
Comment (huaxingao) [reply sql/core/src/test/scala/org/apache/spark/sql/jdbc/JDBCV2Suite.scala:348]: OK. I see what you mean now.

**Review by cloud-fan — APPROVED** [review]

**Review by c21 — APPROVED** [review]

**Comment by cloud-fan** [comment]
thanks, merging to master!

**Pull request closed without merging**